docker/test_env/ethereum_env.go

```diff
@@ -108,265 +127,363 @@ func (b *EthereumNetworkBuilder) WithTest(t *testing.T) *EthereumNetworkBuilder
+	net.SupportsEIP1559 = true
+
+	//TODO when we support multiple participants, we need to modify net so that it contains all the RPC URLs, not just the last one
+
 	return net, rpcProvider, nil
 }
 
-func (b *EthereumNetwork) startPow() (blockchain.EVMNetwork, RpcProvider, error) {
-	if b.ExecutionLayer != ExecutionLayer_Geth {
-		return blockchain.EVMNetwork{}, RpcProvider{}, fmt.Errorf("unsupported execution layer: %s", b.ExecutionLayer)
-	}
-	networkNames, err := b.getOrCreateDockerNetworks()
-	if err != nil {
-		return blockchain.EVMNetwork{}, RpcProvider{}, err
+func (en *EthereumNetwork) startPow() (blockchain.EVMNetwork, RpcProvider, error) {
+	var net blockchain.EVMNetwork
+	var networkNames []string
+	rpcProvider := RpcProvider{
+		privateHttpUrls: []string{},
+		privatelWsUrls:  []string{},
+		publiclHttpUrls: []string{},
+		publicsUrls:     []string{},
 	}
 
-	geth := NewGeth(networkNames, b.setExistingContainerName(ContainerType_Geth)).WithTestLogger(b.t)
-	net, docker, err := geth.StartContainer()
-	if err != nil {
-		return blockchain.EVMNetwork{}, RpcProvider{}, err
-	}
+	for _, p := range en.Participants {
+		if p.ExecutionLayer != ExecutionLayer_Geth {
+			return blockchain.EVMNetwork{}, RpcProvider{}, fmt.Errorf("unsupported execution layer: %s", p.ExecutionLayer)
+		}
+		singleNetwork, err := en.getOrCreateDockerNetworks()
+		if err != nil {
+			return blockchain.EVMNetwork{}, RpcProvider{}, err
+		}
+
+		geth := NewGeth(singleNetwork, *en.ethereumChainConfig, en.setExistingContainerName(ContainerType_Geth)).WithTestLogger(en.t)
+		network, docker, err := geth.StartContainer()
+		if err != nil {
+			return blockchain.EVMNetwork{}, RpcProvider{}, err
+		}
+
+		net = network
+		networkNames = append(networkNames, singleNetwork...)
+
+		containers := EthereumNetworkContainers{
+			{
+				ContainerName: geth.ContainerName,
+				ContainerType: ContainerType_Geth,
+				Container:     &geth.Container,
+			},
+		}
 
-	b.DockerNetworkNames = networkNames
-	b.Containers = EthereumNetworkContainers{
-		{
-			ContainerName: geth.ContainerName,
-			ContainerType: ContainerType_Geth,
-			Container:     &geth.Container,
-		},
+		en.Containers = append(en.Containers, containers...)
+		rpcProvider.privateHttpUrls = append(rpcProvider.privateHttpUrls, docker.HttpUrl)
+		rpcProvider.privatelWsUrls = append(rpcProvider.privatelWsUrls, docker.WsUrl)
+		rpcProvider.publiclHttpUrls = append(rpcProvider.publiclHttpUrls, geth.ExternalHttpUrl)
+		rpcProvider.publicsUrls = append(rpcProvider.publicsUrls, geth.ExternalWsUrl)
 	}
 
-	return net, RpcProvider{
-		privateHttpUrls: []string{docker.HttpUrl},
-		privatelWsUrls:  []string{docker.WsUrl},
-		publiclHttpUrls: []string{geth.ExternalHttpUrl},
-		publicsUrls:     []string{geth.ExternalWsUrl},
-	}, nil
+	en.DockerNetworkNames = networkNames
+	//TODO when we support multiple participants, we need to modify net so that it contains all the RPC URLs, not just the last one
```

Comment:
Is the support issue mainly within our test client, or with setting up the network?